test/e2e/test/elasticsearch/checks_data.go

```diff
@@ -59,6 +59,18 @@ func (dc *DataIntegrityCheck) Init() error {
     }
 }
 `
+	// delete index if running check multiple times
+	indexDeletion, err := http.NewRequest(
+		http.MethodDelete,
+		fmt.Sprintf("/%s", dc.indexName),
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+	// delete the index but ignore errors (e.g. if it did not exist yet)
+	_, _ = esClient.Request(context.Background(), indexDeletion)
```

Comment:
Should we catch and ignore non-existing index errors, but raise any other error here (eg. ES unreachable)?

Reply:
Feels redundant given that we:
* have a separate step that tests connectivity that runs just before that 
* if it is unreachable the test will fail anyway immediately afterwards where we create the index